### ISSUE TITLE
Add support for bulk import method when commissioning hosts

### ISSUE THREAD

**Issue by usbenny** [vmware/terraform-provider-vcf#65]
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [ ] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [ ] Do not leave "+1" or other comments that do not add relevant information or questions.
- [ ] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

The current provider runs all host commissioning tasks in parallel.  Currently, SDDC Manager is limited to supporting 2 host commissioning tasks (https://kb.vmware.com/s/article/92494) and 10 tasks of any type

### Use Case(s)

Commissioning more than 2 hosts in parallel

### Potential Configuration

@dimitarproynov demonstrated on a call the ability to have each host be deployed serially which would solve the issue, however it would take longer to deploy many hosts.

Using the host commission API in SDDC Manager and passing it all hosts that need to be commissioned in one JSON payload would only create one task in SDDC Manager and then it controls the marshalling for commissioning those hosts.

I'm not sure how you would aggregate the hosts in terraform for this to happen though.

### References

KB for max parallel tasks: https://kb.vmware.com/s/article/92494

Host Commission API: https://developer.vmware.com/apis/vcf/latest/hosts/#_usecase_hostCommission

**Comment by raj-chelur** [comment]
> Currently, SDDC Manager is limited to supporting 2 host commissioning tasks (https://kb.vmware.com/s/article/92494) and 10 tasks of any type

This KB is describing a situation where a variety of operations are submitted in parallel.

If there are no other active operations running on SDDC Manager, and you want to commission a large number of hosts, that limit is described in [configmax](https://configmax.esp.vmware.com/guest?vmwareproduct=VMware%20Cloud%20Foundation&release=VMware%20Cloud%20Foundation%205.0&categories=17-0,72-0,73-0,87-0,32-0
):

Concurrent Operations 
ESXi Host Commission 
10
Maximum number of concurrent host commission operations recommended per SDDC manager instance are 10. Maximum number of hosts recommended per commission operation is 40. Note: This limit is supported when this is the only operation type submitted in parallel.

Per SDDC Manager 
Hosts in a Commission Operation 
40
Maximum number of input ESXi hosts for each host commission operation in SDDC manager instance

So, you can commission up to 400 hosts in parallel.

**Comment by usbenny** [comment]
Great, while that's a good start you're still limited to 10 at a time then.  The way the Terraform VCF Provider currently works, at least from what I saw, was by creating a commission task for EACH host defined in the TF.  This was the reason for my suggestion, I've seen plenty of commission operations with over 10 hosts.  Simply need a way to batch them up to get to the 400# you reference.

**Comment by raj-chelur** [comment]
Agree, we need to integrate batching into the terraform provider somehow - @dimitarproynov and @tenthirtyam - thoughts?

I was providing the context for what is possible in SDDC Manager APIs.

**Comment by spacegospod** [comment]
I did some research on this one.
Even now we use the bulk API to commission hosts. The technical challenge here is to get all of the host specs into the same  API call. This is difficult because the individual resources don't know about each other.
We could figure out some sort of utility that orchestrates this.

A cleaner solution would be to define a dedicated resource for managing hosts in bulk
